services/MerchStoreService.ts

```diff
@@ -313,6 +314,13 @@ export default class MerchStoreService {
         throw new NotFoundError('Cannot pickup order at an event that starts in less than 2 days');
       }
 
+      // Verify that this order would not set the pickup event's order count
+      // over the order limit
+      const currentOrderCount = pickupEvent.orders.length;
+      if (currentOrderCount >= pickupEvent.orderLimit) {
+        throw new UserError('Cannot place order with a fully-booked pickup event');
+      }
```

Comment:
nit: Since this is a member-facing error message, let's use something more friendly/less technical like 'This merch pickup event is full! Please choose a different pickup event'

Reply:
Resolved in 895837e.